### PR TITLE
Guard 2073 updating encoding

### PR DESCRIPTION
**Pull request by nathan-skuvault**
**Description**
- Updated the encoding to align with NetSuites new requirements for Token-based Authentication: https://docs.oracle.com/en/cloud/saas/netsuite/ns-online-help/section_158921905537.html
- Added basic comments to clarify time sensitive test for future devs.
- Removed/Updated XML documentation in a few places.

### CODE REVIEW

File: src/NetSuiteTests/OrderTests.cs [hunk 1]
```diff
@@ -60,11 +60,11 @@ public void GivenHugeDefaultPurchaseOrdersPageSize_WhenGetPurchaseOrdersAsyncIsC
 		{
 			Config.SearchPurchaseOrdersPageSize = 200;
 
-			var ex = Assert.Throws< NetSuiteException >( () =>
-			{
-				this._ordersService.GetPurchaseOrdersAsync( DateTime.UtcNow.AddDays( -14 ), DateTime.UtcNow, CancellationToken.None ).Wait();
-			} );
-			ex.Should().BeNull();
+			Assert.DoesNotThrow(() => this._ordersService.GetPurchaseOrdersAsync(
```
Comment (maxim-saltanov-skuvault) [inline src/NetSuiteTests/OrderTests.cs:63]: @nathan-skuvault code styling: need a space after DoesNotThrow( and after .Wait()

File: src/NetSuiteAccess/Services/Soap/NetSuiteSoapService.cs [hunk 2]
```diff
@@ -62,25 +62,25 @@ public NetSuiteSoapService( NetSuiteConfig config )
 
 		private void ConfigureClient( NetSuitePortTypeClient client )
 		{
-			client.Endpoint.Binding.SendTimeout = new TimeSpan( 0, 0, 0, 0, this._config.NetworkOptions.RequestTimeoutMs );
-
-			string subdomain = this._config.Credentials.CustomerId.ToLowerInvariant().Replace("_", "-");
+			client.Endpoint.Binding.SendTimeout = new TimeSpan( 0, 0, 0, 0, this._config.NetworkOptions.RequestTimeoutMs );
+
+			string subdomain = this._config.Credentials.CustomerId.ToLowerInvariant().Replace("_", "-");
```
Comment (maxim-saltanov-skuvault) [inline src/NetSuiteAccess/Services/Soap/NetSuiteSoapService.cs:67]: @nathan-skuvault code styling: need spaces inside Replace method

File: src/NetSuiteAccess/Services/Soap/NetSuiteSoapService.cs [hunk 3]
```diff
@@ -62,25 +62,25 @@ public NetSuiteSoapService( NetSuiteConfig config )
...
-
-			if ( !dataCenter.EndsWith( "/" ) )
-			{
-				return new EndpointAddress( dataCenter + relativeWsPath );
-			}
-			else
-			{
-				return new EndpointAddress( string.Concat( dataCenter.Substring( 0, dataCenter.Length - 1), relativeWsPath ) );
-			}
+		private EndpointAddress GetDataCenterEndpoint( NetSuitePortTypeClient client, string dataCenter)
```
Comment (maxim-saltanov-skuvault) [inline src/NetSuiteAccess/Services/Soap/NetSuiteSoapService.cs:71]: @nathan-skuvault code styling: need a space after "string dataCenter"

File: src/NetSuiteAccess/Services/Soap/NetSuiteSoapService.cs [hunk 4]
```diff
@@ -62,25 +62,25 @@ public NetSuiteSoapService( NetSuiteConfig config )
...
+			var endpoint = client.Endpoint.Address;
+			var relativeWsPath = endpoint.Uri.LocalPath;
+
+			if ( !dataCenter.EndsWith( "/" ) )
+			{
+				return new EndpointAddress( dataCenter + relativeWsPath );
+			}
+			else
+			{
+				return new EndpointAddress( string.Concat( dataCenter.Substring( 0, dataCenter.Length - 1), relativeWsPath ) );
```
Comment (maxim-saltanov-skuvault) [inline src/NetSuiteAccess/Services/Soap/NetSuiteSoapService.cs:82]: @nathan-skuvault code styling: need a space after ".Length - 1"

**Review by maxim-saltanov-skuvault — APPROVED** [review]
all fine (except small code styles fixes)

File: src/NetSuiteAccess/Services/Soap/NetSuiteSoapService.cs (hunk 4 above)
Comment (nathan-skuvault) [reply src/NetSuiteAccess/Services/Soap/NetSuiteSoapService.cs]: Pushed formatting fix!

File: src/NetSuiteAccess/Services/Soap/NetSuiteSoapService.cs (hunk 3 above)
Comment (nathan-skuvault) [reply src/NetSuiteAccess/Services/Soap/NetSuiteSoapService.cs]: Pushed formatting fix!

File: src/NetSuiteAccess/Services/Soap/NetSuiteSoapService.cs (hunk 2 above)
Comment (nathan-skuvault) [reply src/NetSuiteAccess/Services/Soap/NetSuiteSoapService.cs]: Pushed formatting fix!

File: src/NetSuiteTests/OrderTests.cs (hunk 1 above)
Comment (nathan-skuvault) [reply src/NetSuiteTests/OrderTests.cs]: Pushed formatting fix!

**Review by maxim-skuvault — APPROVED** [review]

**Pull request merged**